### PR TITLE
Override default API url in local/staging envs

### PR DESCRIPTION
**Pull request by AndrewDryga**

### CODE REVIEW

File: elixir/apps/web/lib/web/live/sites/new_token.ex [hunk 1]
```diff
@@ -4,12 +4,20 @@ defmodule Web.Sites.NewToken do
 
   def mount(%{"id" => id}, _session, socket) do
     with {:ok, group} <- Gateways.fetch_group_by_id(id, socket.assigns.subject) do
-      {:ok, group} =
-        Gateways.update_group(%{group | tokens: []}, %{tokens: [%{}]}, socket.assigns.subject)
+      {group, env} =
+        if connected?(socket) do
+          {:ok, group} =
+            Gateways.update_group(%{group | tokens: []}, %{tokens: [%{}]}, socket.assigns.subject)
```
Comment (AndrewDryga) [inline elixir/apps/web/lib/web/live/sites/new_token.ex:10]: @bmanifold `mount/3` is called twice: when a static page is generated and then once more when web socket is connected. So behind the scenes if you hit refresh we will insert the token two times. This trick allows to check which kind of render this is and avoid inserting a record before WS is connected. Downside is that container will be empty for a brief moment while it loads.

File: elixir/apps/web/assets/js/hooks.js [hunk 2]
```diff
@@ -9,7 +9,10 @@ Hooks.Copy = {
     this.el.addEventListener("click", (ev) => {
       ev.preventDefault();
 
-      let text = ev.currentTarget.querySelector("[data-copy]").innerHTML.trim();
+      let inner_html = ev.currentTarget.querySelector("[data-copy]").innerHTML.trim();
```
Comment (AndrewDryga) [inline elixir/apps/web/assets/js/hooks.js:12]: `.innerHTML` returns a percent-encoded string, so we copy-pasted code that wasn't able to run if it included any `&` or `<` characters.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -119,12 +132,7 @@ jobs:
           file: ${{ matrix.context }}/Dockerfile
           push: true
           target: ${{ matrix.target }}
-          tags: |
-            ${{ github.ref == 'refs/heads/main' && format('{0}/firezone/{1}:{2}', steps.login.outputs.registry, matrix.image_name, env.MAJOR_VERSION) }}
```
Comment (AndrewDryga) [inline .github/workflows/ci.yml:123]: This wasn't working

File: elixir/apps/web/assets/css/scrollbar.css
Comment (jamilbk) [inline elixir/apps/web/assets/css/scrollbar.css:1]: Curious why the changes to this file? Wasn't this originally broken and fixed by a contributor?

File: elixir/apps/web/assets/js/hooks.js (hunk 2 above)
Comment (jamilbk) [reply elixir/apps/web/assets/js/hooks.js:12]: Good catch!

File: elixir/apps/web/lib/web/live/sites/new_token.ex [hunk 4]
```diff
@@ -65,65 +115,83 @@ defmodule Web.Sites.NewToken do
...
-      --env FIREZONE_TOKEN="#{token}" \\
-      --env FIREZONE_ENABLE_MASQUERADE=1 \\
-      --env FIREZONE_HOSTNAME="`hostname`" \\
-      --env RUST_LOG="warn" \\
-      #{Domain.Config.fetch_env!(:domain, :docker_registry)}/gateway:#{version()}
-    """
+  defp env(token) do
+    api_url_override =
+      if api_url = Domain.Config.get_env(:web, :api_url_override) do
+        # if api_url = false do
```
Comment (jamilbk) [inline elixir/apps/web/lib/web/live/sites/new_token.ex]: Should this be removed?

**Review by jamilbk — APPROVED** [review]
Just some minor comments / suggestions otherwise LGTM

File: elixir/apps/web/assets/css/scrollbar.css
Comment (AndrewDryga) [reply elixir/apps/web/assets/css/scrollbar.css:1]: @jamilbk it was still broken on the latest Chrome because when you hover on the code block 0.5rem bar is shown and the content jumps up and back. I decided to completely disable scrollbars because don't want to deal with all the browsers testing now and we expect people to just copy-paste code anyway.

File: elixir/apps/web/lib/web/live/sites/new_token.ex (hunk 4 above)
Comment (AndrewDryga) [reply elixir/apps/web/lib/web/live/sites/new_token.ex]: Good catch!
